lib/edgehog_device_forwarder/termination_callbacks.ex

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+defmodule EdgehogDeviceForwarder.TerminationCallbacks do
+  use GenServer
+
+  @table :termination_callbacks_table
+
+  @spec add(pid, (-> any)) :: :ok
+  def add(pid, on_close) when is_pid(pid) and is_function(on_close, 0) do
+    GenServer.cast(__MODULE__, {:add, pid, on_close})
```

Comment:
This should be a `:call`, otherwise you don't have any guarantee about the fact the message actually went through (`:cast` will happily return `:ok` if the process you're sending to is dead)